src/test/java/apoc/export/cypher/ExportCypherTest.java

```diff
@@ -27,41 +23,130 @@
  */
 public class ExportCypherTest {
 
-    private static final String EXPECTED_NODES = String.format("begin%n" +
-            "CREATE (:`Foo`:`UNIQUE IMPORT LABEL` {`name`:\"foo\", `UNIQUE IMPORT ID`:0});%n" +
-            "CREATE (:`Bar` {`name`:\"bar\", `age`:42});%n" +
-            "CREATE (:`Bar`:`UNIQUE IMPORT LABEL` {`age`:12, `UNIQUE IMPORT ID`:2});%n" +
-            "commit%n");
-
-    private static final String EXPECTED_SCHEMA = String.format("begin%n" +
-            "CREATE INDEX ON :`Foo`(`name`);%n" +
-            "CREATE CONSTRAINT ON (node:`Bar`) ASSERT node.`name` IS UNIQUE;%n" +
-            "CREATE CONSTRAINT ON (node:`UNIQUE IMPORT LABEL`) ASSERT node.`UNIQUE IMPORT ID` IS UNIQUE;%n" +
-            "commit%n" +
-            "schema await%n");
-
-    private static final String EXPECTED_RELATIONSHIPS = String.format("begin%n" +
-            "MATCH (n1:`UNIQUE IMPORT LABEL`{`UNIQUE IMPORT ID`:0}), (n2:`Bar`{`name`:\"bar\"}) CREATE (n1)-[:`KNOWS`]->(n2);%n" +
-            "commit%n");
-
-    private static final String EXPECTED_CLEAN_UP = String.format("begin%n" +
-            "MATCH (n:`UNIQUE IMPORT LABEL`)  WITH n LIMIT 20000 REMOVE n:`UNIQUE IMPORT LABEL` REMOVE n.`UNIQUE IMPORT ID`;%n" +
-            "commit%n" +
-            "begin%n" +
-            "DROP CONSTRAINT ON (node:`UNIQUE IMPORT LABEL`) ASSERT node.`UNIQUE IMPORT ID` IS UNIQUE;%n" +
-            "commit%n");
-
-    private static final String EXPECTED_NEO4J_SHELL = EXPECTED_NODES + EXPECTED_SCHEMA + EXPECTED_RELATIONSHIPS + EXPECTED_CLEAN_UP;
-
-    private static final String EXPECTED_CYPHER_SHELL = EXPECTED_NEO4J_SHELL
-            .replace(NEO4J_SHELL.begin(), CYPHER_SHELL.begin())
-            .replace(NEO4J_SHELL.commit(),CYPHER_SHELL.commit())
-            .replace(NEO4J_SHELL.schemaAwait(),CYPHER_SHELL.schemaAwait());
-
-    private static final String EXPECTED_PLAIN = EXPECTED_NEO4J_SHELL
-            .replace(NEO4J_SHELL.begin(), PLAIN_FORMAT.begin())
-            .replace(NEO4J_SHELL.commit(),PLAIN_FORMAT.commit())
-            .replace(NEO4J_SHELL.schemaAwait(),PLAIN_FORMAT.schemaAwait());
+    private interface ExpectedOutputs {
+        String nodes();
+        String schema();
+        String relationships();
+        String cleanup();
+        default String all()
+        {
+            return nodes() + schema() + relationships() + cleanup();
+        }
+    }
+
+    private static class CypherOutput implements ExpectedOutputs
+    {
+        @Override
+        public String nodes() {
+            return String.format(":BEGIN%n" +
+                    "CREATE (:`Foo`:`UNIQUE IMPORT LABEL` {`name`:\"foo\", `UNIQUE IMPORT ID`:0});%n" +
```

Comment:
Yeah - but I don't like programmatically generating assertion text in test code. It's hard to read what the assertion should be, and you very quickly end up with test code that has the same behavior (and bugs) as the code you're intending to test.